File: usecase/core/src/main/java/com/hearlers/gateway/port/TokenProviderPort.java
package com.hearlers.gateway.port;

import com.hearlers.api.proto.v1.model.AuthChannel;
import com.hearlers.api.proto.v1.model.Authority;

import java.time.LocalDateTime;

public interface TokenProviderPort {
    boolean validateToken(String token);
    boolean isTokenExpired(String token);
    String getUserId(String token);
    AuthChannel getAuthChannel(String token);
    Authority getAuthority(String token);
    String createToken(String userId, AuthChannel authChannel, Authority authority, LocalDateTime validity);
    long getAccessTokenExpirationTime();
    long getRefreshTokenExpirationTime();
}

File: usecase/core/src/main/kotlin/com/hearlers/gateway/port/TokenProviderPort.kt
package com.hearlers.com.hearlers.gateway.port

import com.hearlers.api.proto.v1.model.AuthChannel
import com.hearlers.api.proto.v1.model.Authority
import java.time.LocalDateTime

interface TokenProviderPort {
    fun validateToken(token: String): Boolean
    fun isTokenExpired(token: String): Boolean
    fun getUserId(token: String): String
    fun getAuthChannel(token: String): AuthChannel
    fun getAuthority(token: String): Authority
    fun createToken(
        userId: String,
        authChannel: AuthChannel,
        authority: Authority,
        validity: LocalDateTime
    ): String
    val accessTokenExpirationTime: Long
    val refreshTokenExpirationTime: Long
}